faraday/proxy.py

```diff
@@ -393,6 +428,208 @@ def testdb_read_worker():
         row = cursor.fetchone()
 
 
+def acceptConnection(server, faraday):
+    conn, addr = server.accept()
+    logger.info("Got connection from {0} on {1}".format(addr, faraday))
+    return conn, addr
+
+
+def closeConnection(conn, addr, faraday):
+    # close the connection
+    logger.info("Closing connection with {0} on {1}".format(addr, faraday))
+    try:
+        conn.close()
+    except socket.error as e:
+        logger.error(e)
+
+
+def extractBytes(data, dataBuffer, unit):
+    # data is a BASE64 string with unknown length, iterate and separete
+    #data = data.decode('base64', 'strict')
+
+    for byte in data:
+        try:
+            byte = struct.unpack("c", byte)[0]
+            dataBuffer[unit].append(byte)
+
+        except:
+            logger.debug("Creating dataBuffer")
+            dataBuffer[unit] = deque([])
+            dataBuffer[unit].append(byte)
+
+
+def receiveData(conn, addr, dataBuffer, unit):
+    while True:
+        try:
+            data = conn.recv(4096)
+        except socket.error as e:
+            logger.error(e)
+            closeConnection(conn, addr, unit)
+            break
+
+        if not data:
+            closeConnection(conn, addr, unit)
+            break
+
+        # Expect BASE64 so decode it
+        extractBytes(data, dataBuffer, unit)
+
+
+def sendData(conn, addr, getDicts, unit, payloadSize):
+    while True:
+        # not sure why I need a delay... otherwise socket closes
+        time.sleep(0.05)
```

Comment:
This was partially addressed in https://github.com/FaradayRF/Faraday-Software/pull/251/commits/90b20a67aea82a58978c63f926bf3760e7498a02 as I removed the sleeps and now the send code is blazing along. Very sure Faraday firmware is limiting here. Or at least UART code.

However, I noticed a bug where we can't connect to the RX port and disconnect without ever sending data to the TX port (so RX is populated). This is a weird bug...